File: app/src/main/java/com/example/rwothoromo/developers/view/GithubUserView.java
package com.example.rwothoromo.developers.view;

import com.example.rwothoromo.developers.model.GithubUser;

import java.util.List;

/**
 * This class represents the GitHub User view interface.
 */
public interface GithubUserView {

    /**
     * Method to act on available GitHub user list.
     *
     * @param githubUsers the GitHub user list
     */
    void githubUsersReady(List<GithubUser> githubUsers);

    /**
     * Method to act on failed GitHub API call.
     */
    void failedDataRetrieval();
}

File: app/src/main/java/com/example/rwothoromo/developers/view/GithubUserView.kt
package com.example.rwothoromo.developers.view

import com.example.rwothoromo.developers.model.GithubUser

/**
 * This class represents the GitHub User view interface.
 */
interface GithubUserView {

    /**
     * Method to act on available GitHub user list.
     *
     * @param githubUsers the GitHub user list
     */
    fun githubUsersReady(githubUsers: List<GithubUser>)

    /**
     * Method to act on failed GitHub API call.
     */
    fun failedDataRetrieval()
}
